logic/multisignature.js

```diff
@@ -520,6 +520,18 @@ Multisignature.prototype.schema = {
  * @returns {transaction} Validated transaction
  */
 Multisignature.prototype.objectNormalize = function(transaction) {
+	if (
```

Comment:
There is only one place when we set this `ready` flag:
https://github.com/LiskHQ/lisk/blob/4e4a344e0313fa21dd58fa14b56e96dd1e78f3db/modules/multisignatures.js#L172
It's calling this function:
https://github.com/LiskHQ/lisk/blob/4e4a344e0313fa21dd58fa14b56e96dd1e78f3db/logic/multisignature.js#L584-L599
The logic there is different. Also, there should be a single place where we set this `ready` flag, `objectNormalize` is not the proper place. But according to the issue we need to set default values only for `ready` and `signatures`. So what just not add here:
```
transaction.ready = false;
transaction.signatures = Array.isArray(transaction.signatures) ? transaction.signatures : [];
```

Reply:
You can not always set `transaction.ready = false` because that might cause offline transactions to get stuck in the transaction pool. If you send a multi-signature transaction with all signatures present and set the ready property to false it will never be removed from the pool.

Reply:
The logic you suggest to use from https://github.com/LiskHQ/lisk/blob/4e4a344e0313fa21dd58fa14b56e96dd1e78f3db/logic/multisignature.js#L584-L599 expects a transaction that does not presents some edge cases as the one mentioned above - though I did thought of using that function done in the objectNormalize makes sure the transaction pool always gets good data -

Reply:
So then the logic needs to be fixed, `transaction.ready` flag should be never set by the user. Maybe weould be nice to set this flag to `true` here? https://github.com/LiskHQ/lisk/blob/4e4a344e0313fa21dd58fa14b56e96dd1e78f3db/logic/multisignature.js#L168

Reply:
It needs to be set for the case of offline transactions, that is the case when a user sends a transaction which includes the signatures. I think we could iterate over the transaction pool logic in a different issue. 

This method you suggest also looks like a good place for implementing this check but I think this PR is related to sanitation or recovery rather than verification of the final transaction.

Reply:
I still think that the approach proposed in this PR is not right, `objectNormalize` is a function for making transactions objects consistent, it should not make any decisions.

Reply:
You've got a good point with regards to `objectNormalize` will figure out an alternative to the current solution.